### PR TITLE
feat: use recipient and icon data for group dms in invites

### LINKED ISSUE #389: GroupChannel.__str__ should use recipients if name is null
### Summary

GroupChannel.__str__ should have the recipients used for the name if its null, like the api

This is blocked until https://github.com/discord/discord-api-docs/issues/4572 is resolved

### What is the feature request for?

The core library

### The Problem

Continuation of #382

One possible solution would be to just return an empty string. Another would be to return "Unnamed" which is what the Discord client displays in this case for the channel. Either way like the error says __str__ should always return a string.

### The Ideal Solution

use the recipients key provided by the API

### The Current Solution

make a raw http request to get the recipients

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary
closes #389
depends on discord/discord-api-docs#4572

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/invite.py [hunk 1]
```diff
@@ -85,6 +86,11 @@ class PartialInviteChannel:
 
             Returns the partial channel's name.
 
+            .. versionchanged:: 2.5
+                If the channel is of type :attr:`ChannelType.group`,
+                returns the name that's rendered by the official client.
+
+
```
Comment (shiftinv) [inline disnake/invite.py:93]: spacing

File: disnake/invite.py [hunk 2]
```diff
@@ -95,17 +101,21 @@ class PartialInviteChannel:
...
 
-    __slots__ = ("id", "name", "type")
+    __slots__ = ("id", "name", "type", "_recipients")
 
     def __init__(self, data: InviteChannelPayload):
         self.id: int = int(data["id"])
         self.name: Optional[str] = data.get("name")
         self.type: ChannelType = try_enum(ChannelType, data["type"])
+        if self.type is ChannelType.group:
+            self._recipients: List[Dict[Literal["username"], str]] = data.get("recipients", [])
```
Comment (shiftinv) [inline disnake/invite.py:111]: `_recipients` being unbound for other channel types seems weird, I suppose it's technically fine though

File: disnake/invite.py [hunk 3]
```diff
@@ -95,17 +101,21 @@ class PartialInviteChannel:
         The partial channel's type.
     """
 
-    __slots__ = ("id", "name", "type")
+    __slots__ = ("id", "name", "type", "_recipients")
 
     def __init__(self, data: InviteChannelPayload):
```
Comment (shiftinv) [inline disnake/invite.py:106]: Could you update the payload type accordingly? Perhaps a new type
```py
class InviteChannel(PartialChannel, total=False):
    name: Optional[str]
    recipients: List[<...>]
```
There's also an `icon` field that's currently unsupported, could add that here or in another PR

Comment (onerandomusername) [reply disnake/invite.py:106]: I don't think icon is documented?

Comment (shiftinv) [reply disnake/invite.py:106]: it is:
image
https://discord.com/developers/docs/resources/channel#channel-object-channel-structure

Comment (onerandomusername) [reply disnake/invite.py]: https://github.com/DisnakeDev/disnake/pull/498/commits/a2ef1f7931c19cdc5fd000e3df1c8d9eff3b7144

File: disnake/invite.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/invite.py]: https://github.com/DisnakeDev/disnake/pull/498/commits/bfbbe7ee8db8707f5f61ced06663d22e570aa24d

File: disnake/invite.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/invite.py:118]: https://github.com/DisnakeDev/disnake/pull/498/commits/bfbbe7ee8db8707f5f61ced06663d22e570aa24d

File: disnake/invite.py [hunk 4]
```diff
@@ -120,6 +140,13 @@ def created_at(self) -> datetime.datetime:
         """:class:`datetime.datetime`: Returns the channel's creation time in UTC."""
         return snowflake_time(self.id)
 
+    @property
+    def icon(self) -> Optional[Asset]:
+        """Optional[:class:`Asset`]: Returns the channel's icon asset if available."""
```
Comment (shiftinv) [inline disnake/invite.py:145]: versionadded

File: disnake/types/channel.py [hunk 5]
```diff
@@ -59,6 +59,12 @@ class PartialChannel(_BaseChannel):
     type: ChannelType
 
 
+class InviteChannel(PartialChannel, total=False):
+    name: Optional[str]
+    recipients: List[Dict[Literal["username"], str]]
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/types/channel.py:64]: which part?

[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/types/channel.py:64]: The `Literal["username"]` part, a separate typeddict would make it easier to extend in the future. Thought something like
```py
class InviteUser(TypedDict):
    username: str

class InviteChannel(PartialChannel, total=False):
    name: Optional[str]
    recipients: List[InviteUser]
```

[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/types/channel.py]: https://github.com/DisnakeDev/disnake/pull/498/commits/44f82f2ad43375acd6c285f5aa1b01a263d44411

**Review by shiftinv — APPROVED** [review]
lgtm

**Pull request merged**